lib/common/formats.js

```diff
@@ -689,11 +689,25 @@ const formats = {
       options,
     });
 
-    const dictionaryString = JSON.stringify(
-      minify ? minifyDictionary(tokens, options.usesDtcg) : tokens,
-      null,
-      2,
-    );
+    let dictionaryString;
+    if (flat) {
+      dictionaryString =
+        '{\n' +
+        dictionary.allTokens
+          .map(function (token) {
+            return `  "${token.name}": ${JSON.stringify(
+              options.usesDtcg ? token.$value : token.value,
+            )}`;
+          })
+          .join(',\n') +
+        '\n}';
```

Comment:
```suggestion
      dictionaryString = JSON.stringify(
        Object.fromEntries(
          dictionary.allTokens.map(token => [
            token.name,
            options.usesDtcg ? token.$value : token.value
          ]),
        ),
        null,
        2
      );
```

Cleaner and more optimized to create the object from the array and stringify that in its entirety rather than stringifying each token value separately and string-building the JSON ourselves. You'll have to format this suggestion though, linting will complain if you commit the suggestion as is.

Reply:
Fixed.